modules/tinymce/src/core/main/ts/api/util/URI.ts

```diff
@@ -36,6 +37,34 @@ export interface URIConstructor {
   parseDataUri: (uri: string) => { type: string; data: string };
 }
 
+interface SafeUriOptions {
+  readonly allow_html_data_urls?: boolean;
+  readonly allow_script_urls?: boolean;
+  readonly allow_svg_data_urls?: boolean;
+}
+
+const safeSvgDataUrlElements = [ 'img', 'video' ];
+
+const blockSvgDataUris = (allowSvgDataUrls: boolean | undefined, tagName?: string) => {
+  // Only allow SVGs by default on images/videos since the browser won't execute scripts on those elements
+  if (Type.isNullable(tagName)) {
+    return true;
+  } else {
+    const allowed = Type.isNullable(allowSvgDataUrls) ? Arr.contains(safeSvgDataUrlElements, tagName) : allowSvgDataUrls;
```

Comment:
Could you flip this so that the top-level `if/else` is `Type.isNullable(allowSvgDataUrls)` and then the ternary is `Type.isNullable(tagName) ? !Arr.contains(...`? I just feel like it'll be a little easier to follow if you keep the tagName-related logic together

Reply:
This is literally copied verbatim from the old code in `SaxParser` :shrug: but yeah we should be able to.

Reply:
Fixed